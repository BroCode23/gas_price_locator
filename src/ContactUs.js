import * as React from 'react';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import IconButton from '@mui/material/IconButton';
import MenuIcon from '@mui/icons-material/Menu';
import { styled, useTheme } from '@mui/material/styles';
import Box from '@mui/material/Box';
import MuiDrawer from '@mui/material/Drawer';
import MuiAppBar from '@mui/material/AppBar';
import List from '@mui/material/List';
import CssBaseline from '@mui/material/CssBaseline';
import Divider from '@mui/material/Divider';
import ChevronLeftIcon from '@mui/icons-material/ChevronLeft';
import ChevronRightIcon from '@mui/icons-material/ChevronRight';
import ListItem from '@mui/material/ListItem';
import ListItemButton from '@mui/material/ListItemButton';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import LocalGasStationIcon from '@mui/icons-material/LocalGasStation';
import BorderColorIcon from '@mui/icons-material/BorderColor';
import MailIcon from '@mui/icons-material/Mail';
import ReportIcon from '@mui/icons-material/Report';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Avatar from '@mui/material/Avatar';
import { Link } from "react-router-dom";

const drawerWidth = 200;

const openedMixin = (theme) => ({
    width: drawerWidth,
    transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.enteringScreen,
    }),
    overflowX: 'hidden',
});

const closedMixin = (theme) => ({
    transition: theme.transitions.create('width', {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    overflowX: 'hidden',
    width: `calc(${theme.spacing(7)} + 1px)`,
    [theme.breakpoints.up('sm')]: {
        width: `calc(${theme.spacing(8)} + 1px)`,
    },
});

const DrawerHeader = styled('div')(({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    padding: theme.spacing(0, 1),
    // necessary for content to be below app bar
    ...theme.mixins.toolbar,
}));

const AppBar = styled(MuiAppBar, {
    shouldForwardProp: (prop) => prop !== 'open',
})(({ theme, open }) => ({
    zIndex: theme.zIndex.drawer + 1,
    transition: theme.transitions.create(['width', 'margin'], {
        easing: theme.transitions.easing.sharp,
        duration: theme.transitions.duration.leavingScreen,
    }),
    ...(open && {
        marginLeft: drawerWidth,
        width: `calc(100% - ${drawerWidth}px)`,
        transition: theme.transitions.create(['width', 'margin'], {
            easing: theme.transitions.easing.sharp,
            duration: theme.transitions.duration.enteringScreen,
        }),
    }),
}));

const Drawer = styled(MuiDrawer, { shouldForwardProp: (prop) => prop !== 'open' })(
    ({ theme, open }) => ({
        width: drawerWidth,
        flexShrink: 0,
        whiteSpace: 'nowrap',
        boxSizing: 'border-box',
        ...(open && {
            ...openedMixin(theme),
            '& .MuiDrawer-paper': openedMixin(theme),
        }),
        ...(!open && {
            ...closedMixin(theme),
            '& .MuiDrawer-paper': closedMixin(theme),
        }),
    }),
);

export default function ContactUs() {
    const theme = useTheme();
    const [open, setOpen] = React.useState(false);

    const handleDrawerOpen = () => {
        setOpen(true);
    };

    const handleDrawerClose = () => {
        setOpen(false);
    };

    return (
        <Box sx={{ display: 'flex' }}>
            <CssBaseline />
            <AppBar position="fixed" open={open}>
                <Toolbar>
                    <IconButton
                        color="inherit"
                        aria-label="open drawer"
                        onClick={handleDrawerOpen}
                        edge="start"
                        sx={{
                            marginRight: 5,
                            ...(open && { display: 'none' }),
                        }}
                    >
                        <MenuIcon />
                    </IconButton>
                    <Typography variant="h6" noWrap component="div">
                        Gas Prices
                    </Typography>
                </Toolbar>
            </AppBar>
            <Drawer variant="permanent" open={open}>
                <DrawerHeader>
                    <IconButton onClick={handleDrawerClose}>
                        {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
                    </IconButton>
                </DrawerHeader>
                <Divider />
                <List>
                    <Link to="/" style={{ textDecoration: "initial", color: "black" }}>
                        <ListItem key="Gas Prices" disablePadding sx={{ display: 'block' }}>
                            <ListItemButton
                                sx={{
                                    minHeight: 48,
                                    justifyContent: open ? 'initial' : 'center',
                                    px: 2.5,
                                }}
                            >
                                <ListItemIcon
                                    sx={{
                                        minWidth: 0,
                                        mr: open ? 3 : 'auto',
                                        justifyContent: 'center',
                                    }}
                                >
                                    <LocalGasStationIcon />
                                </ListItemIcon>
                                <ListItemText primary="Gas Prices" variant="button" sx={{ opacity: open ? 1 : 0 }} />
                            </ListItemButton>
                        </ListItem>
                    </Link>
                    <Link to="/gas-form" style={{ textDecoration: "initial", color: "black" }}>
                        <ListItem key='Submit a Price' disablePadding sx={{ display: 'block' }}>
                            <ListItemButton
                                sx={{
                                    minHeight: 48,
                                    justifyContent: open ? 'initial' : 'center',
                                    px: 2.5,
                                }}
                            >
                                <ListItemIcon
                                    sx={{
                                        minWidth: 0,
                                        mr: open ? 3 : 'auto',
                                        justifyContent: 'center',
                                    }}
                                >
                                    <BorderColorIcon />
                                </ListItemIcon>
                                <ListItemText primary='Submit a Price' sx={{ opacity: open ? 1 : 0 }} />
                            </ListItemButton>
                        </ListItem>
                    </Link>
                    <Link to="/contact" style={{ textDecoration: "initial", color: "black" }}>
                        <ListItem key='Contact Us' disablePadding sx={{ display: 'block' }}>
                            <ListItemButton
                                sx={{
                                    minHeight: 48,
                                    justifyContent: open ? 'initial' : 'center',
                                    px: 2.5,
                                }}
                            >
                                <ListItemIcon
                                    sx={{
                                        minWidth: 0,
                                        mr: open ? 3 : 'auto',
                                        justifyContent: 'center',
                                    }}
                                >
                                    <MailIcon />
                                </ListItemIcon>
                                <ListItemText primary='Contact Us' sx={{ opacity: open ? 1 : 0 }} />
                            </ListItemButton>
                        </ListItem>
                    </Link>
                    <Link to="/report" style={{ textDecoration: "initial", color: "black" }} >
                        <ListItem key='Report' disablePadding sx={{ display: 'block' }}>
                            <ListItemButton
                                sx={{
                                    minHeight: 48,
                                    justifyContent: open ? 'initial' : 'center',
                                    px: 2.5,
                                }}
                            >
                                <ListItemIcon
                                    sx={{
                                        minWidth: 0,
                                        mr: open ? 3 : 'auto',
                                        justifyContent: 'center',
                                    }}
                                >
                                    <ReportIcon />
                                </ListItemIcon>
                                <ListItemText primary='Report' sx={{ opacity: open ? 1 : 0 }} />
                            </ListItemButton>
                        </ListItem>
                    </Link>
                </List>
            </Drawer>
            <Box component="main" sx={{
                flexGrow: 1, left: "64px", right: "0px", position: "fixed", height: "100%", overflowY: "auto", bgcolor: '#E7EBF0', p: 3
            }}>
                <DrawerHeader />
                <Card sx={{ width: "fit-content", margin: "auto" }}>
                    <CardContent>
                        <Box
                            sx={{
                                marginTop: 8,
                                display: 'flex',
                                flexDirection: 'column',
                                alignItems: 'center',
                            }}
                        >
                            <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
                                <ReportIcon />
                            </Avatar>
                            <Typography component="h1" variant="h5">Contact Us at any of the Following</Typography>
                            <Typography variant="body" component="div">Phone: 123-456-7890</Typography>
                            <Typography variant="body" component="div">Mail: 12345 Corporate Hwy</Typography>
                            <Typography variant="body" component="div">Email: mailto@corporate.com</Typography>
                        </Box>
                    </CardContent>
                </Card>
            </Box>
        </Box >
    );
}