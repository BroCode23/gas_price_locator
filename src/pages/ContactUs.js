import * as React from 'react';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';
import ReportIcon from '@mui/icons-material/Report';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import Avatar from '@mui/material/Avatar';

import AppToolbar, { DrawerHeader } from '../components/AppToolbar';

export default function ContactUs() {
    return (
        <Box sx={{ display: 'flex' }}>
            <CssBaseline />
            <AppToolbar />
            <Box component="main" sx={{
                flexGrow: 1, left: "64px", right: "0px", position: "fixed", height: "100%", overflowY: "auto", bgcolor: '#E7EBF0', p: 3
            }}>
                <DrawerHeader />
                <Card sx={{ width: "fit-content", margin: "auto" }}>
                    <CardContent>
                        <Box
                            sx={{
                                marginTop: 8,
                                display: 'flex',
                                flexDirection: 'column',
                                alignItems: 'center',
                            }}
                        >
                            <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
                                <ReportIcon />
                            </Avatar>
                            <Typography component="h1" variant="h5">Contact Us at any of the Following</Typography>
                            <Typography variant="body" component="div">Phone: 123-456-7890</Typography>
                            <Typography variant="body" component="div">Mail: 12345 Corporate Hwy</Typography>
                            <Typography variant="body" component="div">Email: mailto@corporate.com</Typography>
                        </Box>
                    </CardContent>
                </Card>
            </Box>
        </Box >
    );
}