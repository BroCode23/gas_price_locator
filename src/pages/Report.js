import * as React from 'react';
import Box from '@mui/material/Box';
import CssBaseline from '@mui/material/CssBaseline';

import AppToolbar, { DrawerHeader } from '../components/AppToolbar';

export default function Report() {
    return (
        <Box sx={{ display: 'flex' }}>
            <CssBaseline />
            <AppToolbar />
            <Box component="main" sx={{
                flexGrow: 1, left: "64px", right: "0px", position: "fixed", height: "100%", overflowY: "auto", bgcolor: '#E7EBF0', p: 3
            }}>
                <DrawerHeader />
            </Box>
        </Box >
    );
}