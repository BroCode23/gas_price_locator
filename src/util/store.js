import { configureStore } from '@reduxjs/toolkit';
import databaseReducer from './Database';

export default configureStore({
    reducer: {
        database: databaseReducer,
    },
})