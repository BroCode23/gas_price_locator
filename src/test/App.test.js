import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import { green, brown } from "@mui/material/colors";
import { render, unmountComponentAtNode } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

import store from "../util/store";
import GasForm from "../pages/GasForm";
import ContactUs from "../pages/ContactUs";
import Report from "../pages/Report";
import Homepage from "../pages/Homepage";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

test("Renders the App correctly", () => {
  const theme = createTheme({
    palette: {
      primary: {
        main: green[500],
      },
      secondary: {
        main: brown[400],
      },
    },
  });

  act(() => {
    render(
      <ThemeProvider theme={theme}>
        <Provider store={store}>
          <Router>
            <Routes>
              <Route path="/" element={<Homepage />} />
              <Route path="gas-form" element={<GasForm />} />
              <Route path="contact" element={<ContactUs />} />
              <Route path="report" element={<Report />} />
            </Routes>
          </Router>
        </Provider>
      </ThemeProvider>,
      container
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
