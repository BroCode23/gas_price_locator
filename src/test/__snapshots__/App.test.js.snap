// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiBox-root css-k008qs"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-19z1ozs-MuiPaper-root-MuiAppBar-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
    >
      <button
        aria-label="open drawer"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart MuiIconButton-sizeMedium css-p55zm9-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="MenuIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </button>
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-8u39c-MuiTypography-root"
      >
        Gas Prices
      </div>
    </div>
  </header>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-2dum1v-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-12i7wg6-MuiPaper-root-MuiDrawer-paper"
    >
      <div
        className="css-1r9jet7"
      >
        <button
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="ChevronLeftIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.41 7.41 14 6l-6 6 6 6 1.41-1.41L10.83 12z"
            />
          </svg>
        </button>
      </div>
      <hr
        className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
      />
      <ul
        className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      >
        <a
          href="/"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "textDecoration": "initial",
            }
          }
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
            disabled={false}
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LocalGasStationIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m19.77 7.23.01-.01-3.72-3.72L15 4.56l2.11 2.11c-.94.36-1.61 1.26-1.61 2.33 0 1.38 1.12 2.5 2.5 2.5.36 0 .69-.08 1-.21v7.21c0 .55-.45 1-1 1s-1-.45-1-1V14c0-1.1-.9-2-2-2h-1V5c0-1.1-.9-2-2-2H6c-1.1 0-2 .9-2 2v16h10v-7.5h1.5v5c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5V9c0-.69-.28-1.32-.73-1.77zM12 10H6V5h6v5zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Gas Prices
                </span>
              </div>
            </div>
          </li>
        </a>
        <a
          href="/gas-form"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "textDecoration": "initial",
            }
          }
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
            disabled={false}
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="BorderColorIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22 24H2v-4h20v4zM13.06 5.19l3.75 3.75L7.75 18H4v-3.75l9.06-9.06zm4.82 2.68-3.75-3.75 1.83-1.83c.39-.39 1.02-.39 1.41 0l2.34 2.34c.39.39.39 1.02 0 1.41l-1.83 1.83z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Submit a Price
                </span>
              </div>
            </div>
          </li>
        </a>
        <a
          href="/contact"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "textDecoration": "initial",
            }
          }
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
            disabled={false}
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="MailIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4-8 5-8-5V6l8 5 8-5v2z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Contact Us
                </span>
              </div>
            </div>
          </li>
        </a>
        <a
          href="/report"
          onClick={[Function]}
          style={
            Object {
              "color": "black",
              "textDecoration": "initial",
            }
          }
        >
          <li
            className="MuiListItem-root MuiListItem-gutters css-uxa1ks-MuiListItem-root"
            disabled={false}
          >
            <div
              className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1fqi8hc-MuiButtonBase-root-MuiListItemButton-root"
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              role="button"
              tabIndex={0}
            >
              <div
                className="MuiListItemIcon-root css-1ddsiuw-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ReportIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.73 3H8.27L3 8.27v7.46L8.27 21h7.46L21 15.73V8.27L15.73 3zM12 17.3c-.72 0-1.3-.58-1.3-1.3 0-.72.58-1.3 1.3-1.3.72 0 1.3.58 1.3 1.3 0 .72-.58 1.3-1.3 1.3zm1-4.3h-2V7h2v6z"
                  />
                </svg>
              </div>
              <div
                className="MuiListItemText-root css-1cag644-MuiListItemText-root"
              >
                <span
                  className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Report
                </span>
              </div>
            </div>
          </li>
        </a>
      </ul>
    </div>
  </div>
  <main
    className="MuiBox-root css-qr2fxg"
  >
    <div
      className="css-1r9jet7"
    />
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <img
              alt="Local Gas Stations"
              border="3px solid #000"
              src="gas_map.png"
            />
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-1wrgmsj-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
            >
              Cheapest Gas in Area
            </div>
            <h3
              className="MuiTypography-root MuiTypography-h3 css-gepadz-MuiTypography-root"
            >
              $
              2.75
            </h3>
            <div
              className="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
            >
              Exxon
            </div>
            <div
              className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
            >
              11302 Space Center Blvd
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-pwngrv-MuiPaper-root-MuiCard-root"
        >
          <div
            className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <h2
              className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-gi7hi0-MuiTypography-root"
            >
              Local Gas Prices
            </h2>
            <table
              className="MuiTable-root css-rqglhn-MuiTable-root"
              role={null}
            >
              <thead
                className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1181pq8-MuiTableCell-root"
                    scope="col"
                  >
                    Date Submitted
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1181pq8-MuiTableCell-root"
                    scope="col"
                  >
                    Name
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeSmall css-1181pq8-MuiTableCell-root"
                    scope="col"
                  >
                    Address
                  </th>
                  <th
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeSmall css-1gjur2s-MuiTableCell-root"
                    scope="col"
                  >
                    Price Per Gallon
                  </th>
                </tr>
              </thead>
              <tbody
                className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                role={null}
              >
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    16 Mar, 2019
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    HEB Fuel
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    3501 Clear Lake City Blvd
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                  >
                    $2.96
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    16 Mar, 2019
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    Motu 2395 Chevron
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    2395 Clear Lake City Blvd
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                  >
                    $3.30
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    16 Mar, 2019
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    Chevron
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    3407 Genoa Red Bluff Rd
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                  >
                    $3.30
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    16 Mar, 2019
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    Pitstop Express #11
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    11302 Space Center Blvd
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                  >
                    $2.97
                  </td>
                </tr>
                <tr
                  className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
                  role={null}
                >
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    15 Mar, 2019
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    Exxon
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                  >
                    11302 Space Center Blvd
                  </td>
                  <td
                    aria-sort={null}
                    className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                  >
                    $2.75
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
